### DISCUSSION TITLE
Bug Report: Unable to redo last action when transforming selection

### DISCUSSION THREAD

**Discussion by miya-sans** [keiwando/toonsquid#308] [Q&A] [answered]
Device: iPad Pro (12.9 inch) (6th generation)
iOS Version: 16.4.1
Version: 1.8.6

https://github.com/keiwando/toonsquid/assets/42623210/c77040f0-be5c-44db-9427-c045806e768d

https://github.com/keiwando/toonsquid/assets/42623210/c2216e03-d35f-490e-9426-9bc5d30ef5cf

Normal transformations can be undo and redo. However, If you select and then transform, the last action can only be undo and not redo (the redo button is greyed out on the right bar, three finger gesture does no work). 

In addition, if you do multiple actions while selected, undo twice, the selection will reappear, but in its original location, so your selection will be different.

When doing transforming, it’s common to undo and redo to see the changes you are making, so preserving the original selections allows you to make changes to transform without reselecting. At the moment, selecting, transforming, and then undoing will take you back to the state before it was selected. If you select and then paint, there will be a separate instance for each undo.

I don’t know if it’s by design that there is a consistency difference between selecting and then transforming, but something similar to clip studio paint select and transform system would work. Something to preserve the selection of the transform between undos and redos.

Having a way to turn off the transform guidelines, or preserve the transform selection when switching to another tool would also assist in making fine adjustments.

**Comment by DoodlerPete** [comment]
I have found the same issue. I hope this gets sorted. It doesn’t stop the animation process but does add unnecessary time to the process.

**Comment by keiwando** [comment, marked as answer]
Thanks for reporting this! That bug should be fixed now with the 1.9 update.
